### PR TITLE
CLI tools

### LINKED ISSUE #260: Improve instance setup
The current setup script is not working on many setups, makes it hard to create packages for Yunohost (for instance), and does more harm than good: it should be replaces with something else. The decision that was taken on Loomio was to have non-interactive tools with an interactive overlay. How I see this feature in details is:

- have a CLI tool to do a specific task, that accepts some arguments
- if one of these arguments is required but not specified, ask the user to input it

The tools that would be needed to replace the current setup script (we could add more of them after):
- [ ] `plm instance new`
- [ ] `plm user new`

[Loomio debate](https://framavox.org/d/1S4Itx6R/the-setup-script)

Plume version (and/or commit): 0.2.0

### PR DESCRIPTION
**Pull request by BaptisteGelez**
- New sub-crate: plume-cli
- Creates a binary called 'plm'
- Uses clap to parse CLI arguments
- Add a first command: plm instance new (to init the local instance)

Example usage:

```
plm instance new -d foo.bar -n 'Hey'
```

Fix #260

### CODE REVIEW

File: plume-cli/src/instance.rs [hunk 1]
```diff
@@ -0,0 +1,63 @@
+use clap::{Arg, ArgMatches, App, SubCommand};
+use diesel::PgConnection;
```
Comment (igalic) [inline plume-cli/src/instance.rs]: shouldn't this be database agnostic now?

Comment (BaptisteGelez) [reply plume-cli/src/instance.rs]: Oops, forgot it! :grin:

Comment (BaptisteGelez) [reply plume-cli/src/instance.rs]: Fixed

**Comment by BaptisteGelez** [comment]
Will also fix #157 #89 and #209

File: docs/CLI.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+```bash
+plm instance new --private --domain plu.me --name 'My Plume Instance' -l 'CC-BY'
+```
+
+**Arguments:**
+
+- `--domain`, `-d`: the domain name on which your instance will be available.
+- `--name`, `-n`: The name of your instance. It will be displayed on the homepage.
+- `--default-license`, `-l`: the license to use for articles written on this instance, if no other license is explicitely specified. Optional, defaults to CC-0.
```
Comment (igalic) [inline docs/CLI.md]: this is not correct anymore

File: docs/INSTALL.md [hunk 3]
```diff
@@ -133,44 +133,73 @@ This command may be useful if you decided to use a separate database server.
 
 ## Starting Plume
```
Comment (igalic) [inline docs/INSTALL.md]: We should have fixed the info on how to compile Plume with sqlite in #226, why DM i only thinking of this now?

File: plume-cli/src/instance.rs [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        _ => println!("Unknwon subcommand"),
+    }
+}
+
+fn new<'a>(args: &ArgMatches<'a>, conn: &Connection) {
+    let domain = args.value_of("domain").map(String::from)
+        .unwrap_or_else(|| env::var("BASE_URL")
+            .unwrap_or_else(|_| super::ask_for("Domain name")));
+    let name = args.value_of("name").map(String::from).unwrap_or_else(|| super::ask_for("Instance name"));
+    let license = args.value_of("default-license").map(String::from).unwrap_or(String::from("CC-0"));
```
Comment (igalic) [inline plume-cli/src/instance.rs]: we've decided to change the default license

**Comment by igalic** [comment]
we have forgotten to fix up the compile / installation instructions after/with #226

**Comment by BaptisteGelez** [comment]
@igalic yes, I thought about it a few minutes ago too. I will update them right now.

File: docs/CLI.md (hunk 2 above)
Comment (BaptisteGelez) [reply docs/CLI.md]: Yes, but I will change everything in the same PR, so for now I used CC-0.

**Pull request merged**